File: py/__tests__/word_counter_test.py
import unittest
from algorithm.word_counter import word_counter


class TestWordCounter(unittest.TestCase):
    def test_counter(self):
        str_dir = word_counter('stay angry, stay foolish')
        # self.assertEqual(str_dir['stay'], 2)
        # self.assertEqual(str_dir['angry'], 1)
        # self.assertEqual(str_dir['foolish'], 1)
        self.assertDictEqual(str_dir, {'stay': 2, 'angry': 1, 'foolish': 1})

File: js/__tests__/word_counter.test.js
const wordCounter = require('../algorithm/word_counter')

describe('wordCounter', () => {
    it('case 1', () => {
        const dict = wordCounter('stay hungry, stay foolish')
        // 浅比较， 不是深比较, 对象h
        expect(dict).toEqual({
            'stay': 2,
            'hungry': 1,
            'foolish': 1
        })
    })
    it('case empty', () => {
        const dict = wordCounter()
        expect(Object.keys(dict).length).toEqual(0)
    })
})
